src/st/nz0/entity_heart_drop.c

```diff
@@ -1,35 +1,36 @@
 #include "nz0.h"
 
 void EntityHeartDrop(Entity* self) {
```

Comment:
Why not create a .h for EntityHeartDrop since we're updating it in so many places?

Reply:
Because each one has its own `g_CastleFlag` index. Each overlay has a slightly different variant of the function. And I couldn't spot a reliable pattern unfortunately.

Reply:
Each one seems to be + 0x100? Am I missing something?
```
g_CastleFlags[(index >> 3) + 0x100] |= 1 << (index & 7);
```

Reply:
It's not that line but two lines above, where some overlays do ` + 0x118` and then `-= 0x118` a bit later. I could realistically add an ad-hoc `#define` and still use a shared header, which is what I was doing with `CASTLE_FLAG_BANK`. But that plan failed. I need to make more research with overlays that actually use candles and item drops.

The other issue is `index = D_80180***[index]`, which is different in each overlay. It wouldn't be a problem if it was not for WRP and RWRP where the array type is different from other overlays for some reason. Again, it needs more exploration.

This is why `e_misc` is the only C file I did not put it yet as a shared header. I do not think it is a blocker for this PR?